.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseSetExtractor.cs

```diff
@@ -91,25 +111,106 @@ public virtual List<Token> TimeEveryday(string text, DateObject reference)
             var ret = new List<Token>();
             var ers = this.config.TimeExtractor.Extract(text, reference);
 
+            if ((config.Options & DateTimeOptions.TasksMode) != 0)
+            {
+                var ers1 = this.config.TimePeriodExtractor.Extract(text, reference);
+                if (ers.Count == 0 && ers1.Count == 1)
+                {
+                    ers = ers1;
+                }
+            }
+
+            foreach (var er in ers)
+            {
+                var afterStr = text.Substring(er.Start + er.Length ?? 0);
+
+                var beforeStr = text.Substring(0, er.Start ?? 0);
+                var beforeMatch = this.config.EachDayRegex.Match(beforeStr);
+                var startIndexBeforeMatch = beforeMatch.Length + beforeMatch.Index - beforeMatch.Value.TrimStart().Length;
+                if (beforeMatch.Success)
+                {
+                    ret.Add(new Token(startIndexBeforeMatch, er.Start + er.Length ?? 0));
+                }
+
+                var match = this.config.EachDayRegex.Match(afterStr);
+                if (match.Success)
+                {
+                    ret.Add(new Token(er.Start ?? 0, (er.Start + er.Length ?? 0) + match.Length + match.Index));
+                }
+            }
+
+            return ret;
+        }
+
+        public virtual List<Token> DayEveryweek(string text, DateObject reference)
+        {
+            var ret = new List<Token>();
+            var ers = this.config.DateExtractor.Extract(text, reference);
+            if (NumberRecognizer.RecognizeOrdinal(text, config.Culture).Count > 0)
+            {
+                return ret;
+            }
+
+            if (ers.Count != 1)
+            {
+                return ret;
+            }
+
             foreach (var er in ers)
             {
                 var afterStr = text.Substring(er.Start + er.Length ?? 0);
-                if ((string.IsNullOrEmpty(afterStr) || this.config.CheckBothBeforeAfter) && this.config.BeforeEachDayRegex != null)
+
+                var beforeStr = text.Substring(0, er.Start ?? 0);
+
+                // var beforeMatch = this.config.EachUnitRegex.Match(beforeStr);

```

Comment:
Please delete the commented out code if not necessary.

Reply:
removed unused code